### PR TITLE
Update yourtubeViews.rb

### PR DESCRIPTION
**Pull request by baweaver**
Tested with: `./yourtube.rb joshwrightpiano`

Output:

``` text
Video                                                                                                                    | Views                | Upload Date     |
-------------------------------------------------------------------------------------------------------------------------------------------------------------------
Chopin Etude in C minor, Op.25 No.12 "Ocean" - ProPractice by Josh Wright                                                | 590 views            | 1 week ago      |
Scarlatti Sonata in A Major, K39 - ProPractice by Josh Wright                                                            | 451 views            | 1 week ago      |
Chopin Etude in A minor, Op.25 No.11 - ProPractice by Josh Wright                                                        | 603 views            | 1 week ago      |
Technique Series: Key of F - Scales, Arpeggios, Triads, and 7th Chords - ProPractice by Josh Wright                      | 342 views            | 1 week ago      |
Technique Series: Key of Bb - Scales, Arpeggios, Triads, and 7th Chords - ProPractice by Josh Wright                     | 276 views            | 2 weeks ago     |
Technique Series: Key of Eb - Scales, Arpeggios, Triads, and 7th Chords - ProPractice by Josh Wright                     | 553 views            | 3 weeks ago     |
Technique Series: Key of Ab - Scales, Arpeggios, Triads, and 7th chords - ProPractice by Josh Wright                     | 379 views            | 3 weeks ago     |
Technique Series: Key of Db - Scales, Arpeggios, Triads, and 7th Chords - ProPractice by Josh Wright                     | 414 views            | 3 weeks ago     |
Technique Series: Key of Gb - Scales, Arpeggios, Triads, and 7th Chords - ProPractice by Josh Wright                     | 336 views            | 4 weeks ago     |
Technique Series: Key of B - Scales, Arpeggios, Triads, and 7th Chords - ProPractice by Josh Wright                      | 425 views            | 4 weeks ago     |
Technique Series: Key of E - Scales, Arpeggios, Triads, and 7th Chords - ProPractice by Josh Wright                      | 378 views            | 1 month ago     |
Technique Series: Key of A - Scales, Arpeggios, Triads, and 7th Chords - ProPractice by Josh Wright                      | 759 views            | 1 month ago     |
Technique Series: Key of D - Scales, Arpeggios, Triads, and 7th Chords - ProPractice by Josh Wright                      | 542 views            | 1 month ago     |
Technique Series: Key of G - Scales, Arpeggios, Triads, and 7th Chords - ProPractice by Josh Wright                      | 759 views            | 1 month ago     |
Technique Series: Key of C - Scales, Arpeggios, Triads, and 7th Chords - ProPractice by Josh Wright                      | 1,219 views          | 1 month ago     |
Minuet in G Minor from the Notebook of Anna Magdalena Bach - ProPractice by Josh Wright                                  | 947 views            | 1 month ago     |
Schumann Sonata No.2 in G minor, Op.22, iv. Rondo. Presto - Josh Wright                                                  | 467 views            | 1 month ago     |
Schumann Sonata No.2 in G minor, Op.22, iii. Scherzo. Sehr rasch und markiert - Josh Wright                              | 403 views            | 1 month ago     |
Schumann Sonata No.2 in G minor, Op.22, ii. Andantino. Getragen - Josh Wright                                            | 164 views            | 1 month ago     |
Schumann Sonata No.2 in G minor, Op.22, i. So rasch wie möglich - Josh Wright                                            | 346 views            | 1 month ago     |
Debussy Golliwogg's Cakewalk from Children's Corner - ProPractice by Josh Wright                                         | 684 views            | 1 month ago     |
A Simple Trick to Develop Fluid Piano Technique - Josh Wright Piano TV                                                   | 4,639 views          | 2 months ago    |
Henri Dutilleux - Au gré des ondes, Étude - Josh Wright                                                                  | 608 views            | 2 months ago    |
Henri Dutilleux - Au gré des ondes, Hommage à Bach - Josh Wright                                                         | 662 views            | 2 months ago    |
Chopin Mazurka in A minor, Op. 17 No. 4 - Josh Wright                                                                    | 1,418 views          | 2 months ago    |
F. Chopin Ballade No. 1 in G minor, Op. 23  (Josh Wright)                                                                | 1,720 views          | 2 months ago    |
F. Chopin Nocturne in D-flat major, Op. 27 No. 2 (Josh Wright)                                                           | 778 views            | 2 months ago    |
F. Chopin Etude in G-flat major, Op. 10 No. 5  (Josh Wright)                                                             | 885 views            | 2 months ago    |
F. Chopin Etude in E minor, Op. 25 No. 5  (Josh Wright)                                                                  | 10,429 views         | 2 months ago    |
F. Chopin Etude in A minor, Op. 25 No. 11 "Winterwind"  (Josh Wright)                                                    | 1,683 views          | 2 months ago    |
```

### CODE REVIEW

File: yourtubeViews.rb [hunk 1]
```diff
@@ -1,3 +1,5 @@
+#!/usr/bin/env ruby
```
Comment (baweaver) [inline yourtubeViews.rb:1]: This tells the shell what to open this script with, always good practice. In this case, it says to ask the environment where ruby is and to use that.

File: yourtubeViews.rb [hunk 2]
```diff
@@ -5,61 +7,28 @@
...
-begin
-  gem 'mechanize'
-rescue LoadError
-  puts "Installing mechanize gem..."
-  system('gem install mechanize')
-  Gem.clear_paths
-end 
-
-require 'mechanize'
+require 'mechanize' rescue Gem::LoadError System 'gem install mechanize'
```
Comment (baweaver) [inline yourtubeViews.rb:10]: You can inline a rescue if you want

File: yourtubeViews.rb [hunk 3]
```diff
@@ -5,61 +7,28 @@
...
-  username = ARGV[0]
-else
-  puts "Please pass in a username to grab the videos from."
-  puts "Ex. yourtubeViews.rb myUsername"
-  exit
-end
+view_element  = "//div[@class='yt-lockup-meta']/ul/li"
 
-mech = Mechanize.new
+username = ARGV[0] or abort 'Please pass in a username to grab the videos from.'
```
Comment (baweaver) [inline yourtubeViews.rb:17]: if `ARGV[0]` is empty, it'll put that message and exit with a status of `-1` which counts as an error

File: yourtubeViews.rb [hunk 4]
```diff
@@ -5,61 +7,28 @@
...
 
-# The user's Youtube videos page
-page = mech.get("https://www.youtube.com/user/#{username}/videos")
+page = Mechanize.new.get("https://www.youtube.com/user/#{username}/videos")
```
Comment (baweaver) [inline yourtubeViews.rb:19]: Just put it all on one line, easier.

File: yourtubeViews.rb [hunk 5]
```diff
@@ -5,61 +7,28 @@
...
 
 # Grab the video titles
-page.search(title_element).each do |title|
-  titles.push(title.text)
-end
+titles = page.search(title_element).map(&:text)
```
Comment (baweaver) [inline yourtubeViews.rb:22]: Map is your friend

File: yourtubeViews.rb [hunk 6]
```diff
@@ -5,61 +7,28 @@
...
-page.search(view_element).each do |info|
-  info = info.text
-  
-  if (info.include?("views"))
-    views.push(info)
-  else
-   upload_dates.push(info)
- end
-end
+views, upload_dates = *page.search(view_element).map(&:text).partition { |info| info.include? 'views' }
```
Comment (baweaver) [inline yourtubeViews.rb:25]: Partition will partition an array on a condition, in this case whether or not the element has views.

File: yourtubeViews.rb [hunk 7]
```diff
@@ -5,61 +7,28 @@
...
 
 # Headings
-printf("%-50s|%-20s|%-15s|\n", "Video", "Views", "Upload Date")
-88.times {print "-"}
-puts 
+printf "%-120s | %-20s | %-15s |\n", "Video", "Views", "Upload Date"
+puts "-" * 163
```
Comment (baweaver) [inline yourtubeViews.rb:29]: It was a bit narrow, and spaces around the `|` helps for readability.

**Comment by baweaver** [comment]
Quick one over, verified to work. Not a bad first run mate.

File: yourtubeViews.rb (hunk 2 above)
Comment (aradreed) [inline yourtubeViews.rb:10]: Doesn't load the mechanize gem afterwards

Comment (baweaver) [inline yourtubeViews.rb:10]:
``` ruby
require 'mechanize' rescue `gem install mechanize` and Gem.clear_paths
```

**Comment by aradreed** [comment]
Edited my code for some of the suggestions. Thanks

**Pull request closed without merging**